### PR TITLE
sink (pulsar): add Expression Pulsar Validate for topic name and unit test .

### PR DESCRIPTION
**Pull request by yumchina**
### What problem does this PR solve?

validate pulsar topic name and some unit test
Issue Number: ref #9413
#### Tests 

 - Unit test
 

##### Will it cause performance regression or break compatibility?
No
##### Do you need to update user documentation, design documentation or monitoring documentation?
No
### Release note 

None

### CODE REVIEW

File: cdc/sink/dmlsink/mq/dispatcher/topic/expression.go [hunk 1]
```diff
@@ -94,3 +102,30 @@ func (e Expression) Substitute(schema, table string) string {
...
+	// validate the topic expression
+	topicName := string(e)
+
+	if len(topicName) == 0 {
+		return errors.ErrPulsarInvalidTopicExpression.GenWithStackByArgs(
+			"topic name is empty")
+	}
+
+	// if not full name, must be simple name
+	if !pulsarTopicNameREFull.MatchString(topicName) {
```
Comment (asddongmen) [inline cdc/sink/dmlsink/mq/dispatcher/topic/expression.go:117]: Prefer to use `require` to check if a case meet expectation, like:https://github.com/pingcap/tiflow/blob/9a5bc3b5b0c99dfaccf36c9dcae8c6471bdf5c0f/cdc/sink/dmlsink/mq/dispatcher/topic/expression_test.go#L218 . I think it's more clear.

File: cdc/sink/dmlsink/mq/dispatcher/topic/expression_pulsar_test.go [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+			t.Errorf("MatchString(%s) failed X", topicName)
+		}
+	}
+}
+
+func TestPulsarValidate(t *testing.T) {
+
+	for _, topicName := range testPulsarTopicList {
+
+		if err := Expression(topicName).PulsarValidate(); err == nil {
```
Comment (asddongmen) [inline cdc/sink/dmlsink/mq/dispatcher/topic/expression_pulsar_test.go:63]: ditto

File: cdc/sink/dmlsink/mq/dispatcher/topic/expression.go (hunk 1 above)
Comment (yumchina) [reply cdc/sink/dmlsink/mq/dispatcher/topic/expression.go:117]: ok，I will

**Review by asddongmen — APPROVED** [review]

File: cdc/sink/dmlsink/mq/dispatcher/topic/expression.go [hunk 3]
```diff
@@ -14,9 +14,9 @@
 package topic
 
 import (
-	"regexp"
-
 	"github.com/pingcap/tiflow/pkg/errors"
+	"regexp"
```
Comment (asddongmen) [inline cdc/sink/dmlsink/mq/dispatcher/topic/expression.go:18]: Keep the import packages like:
```go
import (
    "regexp"
    "strings"
  
    "github.com/pingcap/tiflow/pkg/errors"
)
```
The internal packages should be in the above of all imported packages.

Comment (yumchina) [reply cdc/sink/dmlsink/mq/dispatcher/topic/expression.go:17]: fixed

**Review by 3AceShowHand — APPROVED** [review]

**Pull request merged**